codegen-core/src/main/kotlin/software/amazon/smithy/rust/codegen/core/smithy/generators/StructureGenerator.kt

```diff
@@ -133,7 +133,8 @@ open class StructureGenerator(
         if (accessorMembers.isEmpty()) {
             return
         }
-        writer.rustBlock("impl $name") {
+        val lifetimes = lifetimeDeclaration()
```

Comment:
nit: How about we call this the `genericParams`? Just to be in line with the official [specs](https://doc.rust-lang.org/stable/reference/items/implementations.html).

Reply:
These are specifically lifetimes (from [the docs](https://doc.rust-lang.org/stable/reference/items/generics.html) you linked)